esp-rtos/src/esp_radio/timer_queue.rs

```diff
@@ -158,29 +172,21 @@ impl TimerQueue {
 
             self.inner.with(|q| {
                 let props = current_timer.properties(q);
-                // Set this AFTER the callback so that the callback doesn't leave us in an unknown
-                // "queued?" state.
-                props.enqueued = false;
```

Comment:
If a timer callback ends up re-arming itself, `enqueued` here is `true`. Setting it to false, and enqueueing the timer can break the queue.